### PR TITLE
fix(metadata): warn devs when saving data longer than column holds

### LINKED ISSUE #10861: Add warning if metastring exceeded column length
Without `STRICT_TRANS_TABLES` MySQL truncates metastrings to fit the `text` column. This is problematic when metastring's integrity is important, e.g. a long serialized string. At least a warning would help debug issues that arise from that.

### PR DESCRIPTION
**Pull request by raghukul01**
Fixes #10861

### CODE REVIEW

File: engine/classes/Elgg/Database/MetadataTable.php [hunk 1]
```diff
@@ -123,7 +123,10 @@ function create($entity_guid, $name, $value, $value_type = '', $owner_guid = 0,
 		}
 	
 		$access_id = (int) $access_id;
-	
+		
+		if (strlen($value) >= 50) {
```
Comment (hypeJunction) [inline engine/classes/Elgg/Database/MetadataTable.php]: What's 50? 50 characters?

Comment (hypeJunction) [inline engine/classes/Elgg/Database/MetadataTable.php]: Add `const MAX_VALUE_LENGTH = 65535;` and then compare `strlen($value) >= self::MAX_VALUE_LENGTH`

**Review by hypeJunction — CHANGES_REQUESTED** [review]

File: engine/classes/Elgg/Database/MetadataTable.php [hunk 2]
```diff
@@ -123,7 +126,10 @@ function create($entity_guid, $name, $value, $value_type = '', $owner_guid = 0,
 		}
 	
 		$access_id = (int) $access_id;
-	
+		
+		if (strlen($value) >= self::MAX_VALUE_LENGTH) {
```
Comment (mrclay) [inline engine/classes/Elgg/Database/MetadataTable.php]: `>`, not `>=`

File: engine/classes/Elgg/Database/MetadataTable.php [hunk 3]
```diff
@@ -123,7 +126,10 @@ function create($entity_guid, $name, $value, $value_type = '', $owner_guid = 0,
...
+			elgg_log($name, 'WARNING');
```
Comment (mrclay) [inline engine/classes/Elgg/Database/MetadataTable.php]: Better warning: `"Metadata '$name' is above the MySQL TEXT size limit and may be truncated."`

File: engine/classes/Elgg/Database/MetadataTable.php [hunk 4]
```diff
@@ -41,6 +41,9 @@ class MetadataTable {
 	
 	/** @var string */
 	protected $table;
+	
+	//this constant checks that column size is not exceeded
```
Comment (mrclay) [inline engine/classes/Elgg/Database/MetadataTable.php]: Constants don't check anything. Better would be to name the constant more descriptively so you don't need a comment: `MYSQL_TEXT_BYTE_LIMIT`

**Pull request closed without merging**